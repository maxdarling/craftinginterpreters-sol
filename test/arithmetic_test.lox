// Test basic arithmetic operations
print "=== Arithmetic Operations Tests ===";

// Addition
print 1 + 2; // 3
print 1.5 + 2.5; // 4.0
print "hello" + " " + "world"; // "hello world"

// Subtraction
print 5 - 3; // 2
print 10.5 - 2.5; // 8.0

// Multiplication
print 3 * 4; // 12
print 2.5 * 4; // 10.0

// Division
print 12 / 3; // 4.0
print 15 / 4; // 3.75
print 7 / 2; // 3.5

// Mixed operations with precedence
print 2 + 3 * 4; // 14 (not 20)
print (2 + 3) * 4; // 20
print 10 - 2 * 3; // 4 (not 24)
print (10 - 2) * 3; // 24

// Negative numbers
print -5; // -5
print -(-5); // 5
print 5 + -3; // 2
print 5 - -3; // 8

// Zero operations
print 0 + 5; // 5
print 5 + 0; // 5
print 0 * 5; // 0
print 5 * 0; // 0
print 5 / 1; // 5.0

print "Arithmetic tests completed";
