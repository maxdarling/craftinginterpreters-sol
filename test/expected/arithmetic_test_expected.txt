=== Arithmetic Operations Tests ===
3
4
hello world
2
8
12
10
4
3.75
3.5
14
20
4
24
-5
5
2
8
5
5
0
0
5
Arithmetic tests completed
